asllib/doc/ASLTypingReference.tex

```diff
@@ -2895,260 +3395,397 @@ \subsection{Example}
-  \item One of the following applies:
-    \begin{itemize}
-      \item $\vli$ does not contains any duplicate enumeration literal.
-      \item an error ``\texttt{Already declared identifier}'' is raised;
-    \end{itemize}
-  \item \texttt{new\_ty} is the enumeration type $\tty$.
+  \item determining that $\vli$ does not contain duplicates yields $\True$ \ProseOrTypeError;
+  \item determining that none of the labels in $\vli$ is declared in the global environment
+  yields $\True$ \ProseOrTypeError;
+  \item $\newty$ is the enumeration type $\tty$.
 \end{itemize}
 
 \subsection{Example}
 In the following example, all the uses of enumeration types are valid:
 \VerbatimInput{../tests/ASLTypingReference.t/TypingRule.TEnumDecl.asl}
 
-
-
 \CodeSubsection{\TEnumDeclBegin}{\TEnumDeclEnd}{../Typing.ml}
 
+\begin{emptyformal}
+\subsection{Formally}
+\begin{mathpar}
+\inferrule{
+  \tododefine{check\_no\_duplicates}(\vli) \typearrow \True \OrTypeError\\
+  \vl \in \vli: \tododefine{check\_var\_not\_in\_genv}(\tenv, \vl) \typearrow \True \OrTypeError
+}{
+  \annotatetype{\True, \tenv, \TEnum(\vli)} \typearrow \TEnum(\vli)
+}
+\end{mathpar}
+\end{emptyformal}
 
 \isempty{\subsection{Comments}}
 
-
 \section{TypingRule.TRecordExceptionDecl \label{sec:TypingRule.TRecordExceptionDecl}}
 
 \subsection{Prose}
-Annotating a type~$\tty$ in an environment~\texttt{env} results in a
-rewritten type~\texttt{new\_ty} and one of the following applies:
+All of the following apply:
 \begin{itemize}
-  \item All of the following apply:
-    \begin{itemize}
-      \item $\tty$ is the record type with fields $\fields$;
-      \item This is the declaration of $\tty$;
-      \item One of the following applies:
-        \begin{itemize}
-          \item $\fields$ does not contains any duplicate field.
-          \item an error ``\texttt{Already declared identifier}'' is raised;
-        \end{itemize}
-      \item \texttt{fields'} is the result of annotating each type in
-        $\fields$;
-      \item \texttt{new\_ty} is the record type with fields \texttt{fields'};
-    \end{itemize}
-  \item All of the following apply:
-    \begin{itemize}
-      \item $\tty$ is the exception type with fields $\fields$;
-      \item This is the declaration of $\tty$;
-      \item One of the following applies:
-        \begin{itemize}
-          \item $\fields$ does not contains any duplicate field.
-          \item an error ``\texttt{Already declared identifier}'' is raised;
-        \end{itemize}
-      \item \texttt{fields'} is the result of annotating each type in
-        $\fields$;
-      \item \texttt{new\_ty} is the exception type with fields \texttt{fields'};
-    \end{itemize}
+  \item $\tty$ is either a record type or an exception type, corresponding to its AST label $L$;
+  \item the list of fields of $\tty$ is $\fields$;
+  \item this is the declaration of $\tty$;
+  \item $\fields$ is a list of identifier-type pairs $(\vx_i, \vt_i)$, for $i=1..k$;
```

Comment:
identifier-type pairs, meaning pairs where the first element is an identifier and the second one a type?